code/game/objects/structures/fitness.dm

```diff
@@ -0,0 +1,118 @@
+	flick("[icon_state]2", src)
+	playsound(src.loc, pick(src.hit_sounds), 25, 1, -1)
+
+/obj/structure/punching_bag/wizard
+	icon_state = "punchingbagwizard"
+	desc = "It has a picture of a weird wizard on it."
+
+/obj/structure/punching_bag/syndie
+	icon_state = "punchingbagsyndie"
+	desc = "It has a picture of a mean ol' syndicate on it."
+
+/obj/structure/punching_bag/captain
+	icon_state = "punchingbagcaptain"
+	desc = "It has a picture of a dumb looking station captain on it."
+
+/obj/structure/punching_bag/clown
+	name = "clown bop bag"
+	desc = "A bop bag in the shape of a goofy clown."
+	icon_state = "bopbag"
+
+/obj/structure/punching_bag/clown/attack_hand(mob/user as mob)
+	flick("[icon_state]2", src)
+	playsound(src.loc, pick(src.hit_sounds), 25, 1, -1)
+	playsound(src.loc, 'sound/items/bikehorn.ogg', 50, 1, -1)
+
+/obj/structure/stacklifter
+	name = "Weight Machine"
+	desc = "Just looking at this thing makes you feel tired."
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "fitnesslifter"
+	density = 1
+	anchored = 1
+
+/obj/structure/stacklifter/attack_hand(mob/user as mob)
+	if(in_use)
+		user << "Its already in use - wait a bit."
+		return
+	else
+		in_use = 1
+		icon_state = "fitnesslifter2"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/lifts = 0
+		while (lifts++ < 6)
+			if (user.loc != src.loc)
+				break
+			sleep(3)
+			animate(user, pixel_y = -2, time = 3)
+			sleep(3)
+			animate(user, pixel_y = -4, time = 3)
+			sleep(3)
+			playsound(user, 'sound/effects/spring.ogg', 60, 1)
+
+		playsound(user, 'sound/machines/click.ogg', 60, 1)
+		in_use = 0
+		user.pixel_y = 0
+		var/finishmessage = pick("You feel stronger!","You feel like you can take on the world!","You feel robust!","You feel indestructible!")
+		icon_state = "fitnesslifter"
+		to_chat(user, "<span class='notice'>[finishmessage]</span>")
+
+/obj/structure/weightlifter
+	name = "Weight Machine"
+	desc = "Just looking at this thing makes you feel tired."
+	icon = 'icons/obj/fitness.dmi'
+	icon_state = "fitnessweight"
+	density = 1
+	anchored = 1
+
+/obj/structure/weightlifter/attack_hand(mob/user as mob)
+	if(in_use)
+		user << "Its already in use - wait a bit."
+		return
+	else
+		in_use = 1
+		icon_state = "fitnessweight-c"
+		user.dir = SOUTH
+		user.Stun(4)
+		user.loc = src.loc
+		var/image/W = image('icons/obj/fitness.dmi',"fitnessweight-w")
+		W.layer = MOB_LAYER + 0.1
+		overlays += W
+		var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+		user.visible_message("<B>[user] is [bragmessage]!</B>")
+		var/reps = 0
+		user.pixel_y = 5
+		while (reps++ < 6)
```

Comment:
legit, this is
EXACTLY
what a for loop is for
a while loop is ONLY for when you don't know when it'll end